#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_fix_256119_and_turn_on_sgml_mode.dpatch by Adam Di Carlo <aph@debian.org>.
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: * docbook-4.3/dbpoolx.mod: xml-base.attrib needs to be properly
## DP:   protected from SGML; note the XML and SGML versions of these files
## DP:   stand in for each other
## DP:
## DP: * docbook-4.3/htmltblx.mod: rules made into CDATA attribute rather than
## DP:   an enumeration list
## DP:
## DP: <URL:http://www.oasis-open.org/archives/docbook/200406/msg00027.html>
## DP:
## DP: * docbook-4.{3,4}/htmltblx.mod: include tag minimization entity for
## DP:   cross-function with SGML
## DP:
## DP: <URL:http://bugs.debian.org/256119>
## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.4-1>
## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook-xml/current/changelog#versionversion4.4-5>

@DPATCH@
diff -urNad docbook-xml-4.5~/docbook-4.3/dbpoolx.mod docbook-xml-4.5/docbook-4.3/dbpoolx.mod
--- docbook-xml-4.5~/docbook-4.3/dbpoolx.mod	2007-03-25 23:53:00.000000000 +0200
+++ docbook-xml-4.5/docbook-4.3/dbpoolx.mod	2007-03-25 23:53:00.000000000 +0200
@@ -633,6 +633,9 @@
 
 <!-- xml:base: base URI -->
 
+<![%sgml.features;[
+<!ENTITY % xml-base.attrib "">
+]]>
 <!ENTITY % xml-base.attrib
 	"xml:base	CDATA		#IMPLIED">
 
diff -urNad docbook-xml-4.5~/docbook-4.3/htmltblx.mod docbook-xml-4.5/docbook-4.3/htmltblx.mod
--- docbook-xml-4.5~/docbook-4.3/htmltblx.mod	2004-03-31 13:20:20.000000000 +0200
+++ docbook-xml-4.5/docbook-4.3/htmltblx.mod	2007-03-25 23:53:00.000000000 +0200
@@ -63,6 +63,9 @@
   >
 
 <!-- Does not contain lang or dir because they are in %common.attribs -->
+<![%sgml.features;[
+<!ENTITY % i18n "">
+]]>
 <!ENTITY % i18n
  "xml:lang    NMTOKEN        #IMPLIED"
   >
@@ -92,11 +95,11 @@
   "valign     (top|middle|bottom|baseline) #IMPLIED"
   >
 
-<!ELEMENT colgroup (col)*>
-<!ELEMENT col      EMPTY>
-<!ELEMENT tr       (th|td)+>
-<!ELEMENT th       (%para.char.mix; | %tabentry.mix;)*>
-<!ELEMENT td       (%para.char.mix; | %tabentry.mix;)*>
+<!ELEMENT colgroup %ho; (col)*>
+<!ELEMENT col      %ho; EMPTY>
+<!ELEMENT tr       %ho; (th|td)+>
+<!ELEMENT th       %ho; (%para.char.mix; | %tabentry.mix;)*>
+<!ELEMENT td       %ho; (%para.char.mix; | %tabentry.mix;)*>
 
 <!ATTLIST colgroup
   %attrs;
@@ -206,7 +209,7 @@
     summary     CDATA          #IMPLIED
     width       CDATA        #IMPLIED
     border      CDATA        #IMPLIED
-    rules       (none | groups | rows | cols | all)      #IMPLIED
+    rules       CDATA        #IMPLIED
     cellspacing CDATA        #IMPLIED
     cellpadding CDATA        #IMPLIED
     align       (left|center|right)   #IMPLIED
diff -urNad docbook-xml-4.5~/docbook-4.4/htmltblx.mod docbook-xml-4.5/docbook-4.4/htmltblx.mod
--- docbook-xml-4.5~/docbook-4.4/htmltblx.mod	2005-01-27 14:57:19.000000000 +0100
+++ docbook-xml-4.5/docbook-4.4/htmltblx.mod	2007-03-25 23:53:00.000000000 +0200
@@ -63,6 +63,9 @@
   >
 
 <!-- Does not contain lang or dir because they are in %common.attribs -->
+<![%sgml.features;[
+<!ENTITY % i18n "">
+]]>
 <!ENTITY % i18n
  "xml:lang    NMTOKEN        #IMPLIED"
   >
@@ -92,11 +95,11 @@
   "valign     (top|middle|bottom|baseline) #IMPLIED"
   >
 
-<!ELEMENT colgroup (col)*>
-<!ELEMENT col      EMPTY>
-<!ELEMENT tr       (th|td)+>
-<!ELEMENT th       (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
-<!ELEMENT td       (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
+<!ELEMENT colgroup %ho; (col)*>
+<!ELEMENT col      %ho; EMPTY>
+<!ELEMENT tr       %ho; (th|td)+>
+<!ELEMENT th       %ho; (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
+<!ELEMENT td       %ho; (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
 
 <!ATTLIST colgroup
   %attrs;
